### PR TITLE
Add support for ROS2

### PR DESCRIPTION
**Pull request by lreiher**
- ROS2 node of `mqtt_client` is added
- ROS1 nodelet still exists
- both versions lie next to each other on the same branch
- package can be built either via `catkin` (ROS1) or `colcon` (ROS2)
- README has instructions for both versions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,7 +26,11 @@ The *mqtt_client* package provides a ROS nodelet that enables connected ROS-base
...
 
 ```bash
+# ROS 1
 sudo apt install ros-noetic-mqtt-client
+
+# ROS 2
+sudo apt install ros-humble-mqtt-client
 ```
 
 If you would like to install *mqtt_client* from source, simply clone this repository into your ROS workspace. All dependencies that are listed in the ROS [`package.xml`](package.xml) can be installed using [*rosdep*](http://wiki.ros.org/rosdep).
```
[the remark this answers is not in the document]
Comment (lreiher) [reply README.md:34]: Added.

[the remark this answers is not in the document]
Comment (lreiher) [reply README.md:34]: Added.

File: README.md [hunk 2]
```diff
@@ -110,55 +118,83 @@ Note that the *mqtt_client* successfully connected to the broker and also echoed
...
-```bash
-roslaunch mqtt_client standalone.launch load_params:=false
+# ROS 2
+ros2 launch mqtt_client standalone.launch.ros2.xml params_file:="</PATH/TO/PARAMS.YAML>"
 ```
 
-In order to exploit the benefits of *mqtt_client* being a nodelet, load the nodelet to your own nodelet manager shared with other nodelets.
+In order to exploit the benefits of *mqtt_client* being a ROS 1 nodelet, load the nodelet to your own nodelet manager shared with other nodelets.
 
 ### Configuration
```
[the remark this answers is not in the document]
Comment (lreiher) [reply README.md:206]: I have added an error message if a parameter cannot be loaded with the expected data type, but still exists on the parameter server.

File: README.md [hunk 3]
```diff
@@ -110,55 +118,83 @@ Note that the *mqtt_client* successfully connected to the broker and also echoed
...
+ros2 topic echo /pong/primitive
+```
+
+```bash
+# 5th terminal: publish primitive MQTT message to pingpong/primitive directly using mosquitto_pub
 docker run --rm --network host eclipse-mosquitto mosquitto_pub -h localhost -t "pingpong/primitive" --repeat 20 --repeat-delay 1 -m 69
 ```
 
-If everything works as expected, the second terminal should print a message at 1Hz, while the third terminal should print two different messages at 1Hz.
+If everything works as expected, the second terminal should print a message at 1Hz, while the fourth terminal should print two different messages at 1Hz.
```
[the remark this answers is not in the document]
Comment (lreiher) [reply README.md:179]: Good catch, thank you. A return statement wasn't where it was supposed to be.

**Review by bastilam — APPROVED** [review]

**Comment by lreiher** [comment]
Currently blocked by https://github.com/ros/rosdistro/pull/36406.

**Review by lreiher — COMMENTED** [review]
Should consider whether ROS2 Galactic can also be supported.
https://github.com/ika-rwth-aachen/mqtt_client/issues/19

**Review by lreiher — COMMENTED** [review]
Should consider whether ROS2 Galactic can also be supported.
https://github.com/ika-rwth-aachen/mqtt_client/issues/19

**Comment by lreiher** [comment]
Not going to include ROS2 Galactic support, since it's already EOL since 2022.

**Pull request merged**